integration_tests/src/main/python/repart_test.py

```diff
@@ -23,6 +23,17 @@
 nested_scalar_mark=pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/1459")
 @pytest.mark.parametrize('data_gen', [pytest.param((StructGen([['child0', DecimalGen(7, 2)]]),
                                                     StructGen([['child1', IntegerGen()]])), marks=nested_scalar_mark),
+                                      # left_struct(child0 = 4 level nested, child1 = Int)
+                                      # right_struct(child0 = 4 level nested, child1 = missing)
+                                      (StructGen([['child0', StructGen([['child0', StructGen([['child0', StructGen([['child0',
```

Comment:
Do we have to worry about [SPARK-35108](https://issues.apache.org/jira/browse/SPARK-35108)?  We may need to vary the number of fields at each level so there's no accidental aliasing to trigger that Spark bug.

Reply:
Thanks for this. I have updated the PR to have a different number of children at each level for tests testing against spark < 3.1. The fix for the bug went in 3.0.3 I believe

Reply:
I see the new `nested_struct` was added, but it's not used here?  Seems like most structs generated for this test only have one child.

Reply:
so this test will skip for versions < 311. The bug SPARK-35108 was fixed for 3.0.3+. I could've added it to this test like I did for the others too but this one needs a tuple of structs for left and right and I didn't want to compromise the readability unless it is needed. 

I could still do it if you want me to